tests/test_models.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Tests for the `django-user-tasks` models module."""
+
+from __future__ import absolute_import, unicode_literals
+
+import logging
+from uuid import uuid4
+
+from django.contrib.auth.models import User
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.test import TestCase
+
+import pytest
+
+from user_tasks.exceptions import TaskCanceledException
+from user_tasks.models import UserTaskArtifact, UserTaskStatus
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TestUserTaskStatus(TestCase):
+    """Tests of the UserTask model."""
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = User.objects.create_user('test_user', 'test@example.com', 'password')
+
+    def test_cancel_parent(self):
+        """Canceling a container status should also cancel its children."""
+        parent = self._status(is_container=True, state=UserTaskStatus.PENDING)
+        status = self._status(parent=parent, state=UserTaskStatus.PENDING)
+        parent.cancel()
+        status.refresh_from_db()
+        assert parent.state == UserTaskStatus.CANCELED
```

Comment:
pytest has [special handling](http://doc.pytest.org/en/latest/assert.html) of assert statements that gives pretty useful output on failures (usually better than the unittest methods).  Has the added benefits of being more readable and not needing to remember a dozen or so different but related methods for different types of assertions.